packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncTimeoutOptions.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2023 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.kotlin.mongodb.sync
+
+import kotlin.time.Duration
+
+/**
+ * The configured timeouts for various aspects of the sync connection between synchronized realms
+ * and App Services.
```

Comment:
We should probably make a reference to `SyncTimeoutOptionsBuilder` to make it obvious how to configure this. Or maybe just a:
```
* @see [io.realm.kotlin.mongodb.AppConfiguration.Builder.syncTimeouts]
``` 
Just like for the builder.